### ISSUE TITLE
Vim mode for the editor

### ISSUE THREAD

**Issue by anothernode** [TryGhost/Ghost-Desktop#302]
A Vim mode for the editor is the missing piece that would, in my book at least, transcend Ghost/Ghost Desktop to absolute perfection.

Would that be possible to do at some point? Oh my god, it would make me so happy 😆

**Comment by felixrieseberg** [comment]
Let me turn the question around: If I added an extensions feature that'd give you full access to the editor, would you build the `ghost-vim` extension?

**Comment by kevinansfield** [comment]
For the current CodeMirror based markdown editor this may not be too difficult if you can inject code into the admin thanks to https://codemirror.net/demo/vim.html

Things will likely get more complex once the Mobiledoc based rich-text/card editor lands.

**Comment by anothernode** [comment]
@felixrieseberg: I'm not sure if my skills and time resources are sufficient to do that but I'd certainly love to try!

@kevinansfield: I'm a bit worried about this Mobiledoc rich text editor stuff. Will I still be able to use Markdown to write stuff in Ghost? That is one of the more important things I love so much about it!

**Comment by kevinansfield** [comment]
> I'm a bit worried about this Mobiledoc rich text editor stuff. Will I still be able to use Markdown to write stuff in Ghost?

Yes! A post will still be able consist of a single markdown card if all you want to write in is markdown, we certainly don't want to take that away from you 😄

**Comment by anothernode** [comment]
That's a relief! 😅

If you really add this extension feature I'd give it a shot, but I can't promise anything.

[Quiver](http://happenapps.com/#quiver) uses an editor system that sound's similar to what you are planning. They support different types of "cells" in one note. A cell can be Markdown, LaTeX, Code or Rich Text. For the first three Quiver supports editing in Vim mode, but not for the Rich Text cells.
